sklearn/inspection/_partial_dependence.py

```diff
@@ -321,6 +325,13 @@ def partial_dependence(
             `kind='average'` will be the new default. It is intended to migrate
             from the ndarray output to :class:`~sklearn.utils.Bunch` output.
 
+    is_categorical : array-like of boolean
```

Comment:
Speaking with @ogrisel, we think that we could be consistent with `HistGradientBoosting` and the parameter `categorical_features`. The parameter will be decorrelated from `features` and provide the categorical features present in `X`.

Reply:
@madhuracj I wanted to look at the example. I forgot to push into the branch (or make a PR).
You can check my implementation and compare it with yours: https://github.com/glemaitre/scikit-learn/tree/pdp_categorical_proto

This would be my review most probably.

Reply:
Looks like we both worked on changing, `is_categorical` to `categorical_features`. 
I guess I will revert 8b9263b and 03f7c9c and force push, so it wouldn't be conflicting when you are done with your changes?

Reply:
as you wish